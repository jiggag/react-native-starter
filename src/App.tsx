import React, { useCallback, useEffect } from 'react';
import notifee, { EventType, IOSAuthorizationStatus } from '@notifee/react-native';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { QueryClient, QueryClientProvider } from 'react-query';
import { Detail } from 'screens/Detail';
import { Main } from 'screens/Main';

const queryClient = new QueryClient();

if (__DEV__) {
  // eslint-disable-next-line import/no-extraneous-dependencies
  import('react-query-native-devtools').then(({ addPlugin }) => {
    addPlugin({ queryClient });

    return true;
  }).catch((err) => {
    console.error(err);
  });
}

const Stack = createNativeStackNavigator();

export function App() {
  const checkApplicationPermission = useCallback(async () => {
    const settings = await notifee.requestPermission();
    if (settings.authorizationStatus >= IOSAuthorizationStatus.AUTHORIZED) {
      console.log('Permission settings:', settings);
    } else {
      console.log('User declined permissions');
    }
  }, []);

  useEffect(() => {
    checkApplicationPermission();

    return notifee.onForegroundEvent(({ type, detail }) => {
      switch (type) {
        case EventType.DISMISSED:
          console.log('User dismissed notification', detail.notification);
          break;
        case EventType.PRESS:
          console.log('User pressed notification', detail.notification);
          break;
        default:
          break;
      }
    });
  }, [checkApplicationPermission]);

  return (
    <QueryClientProvider client={queryClient}>
      <NavigationContainer>
        <Stack.Navigator>
          <Stack.Screen name="Main" component={Main} />
          <Stack.Screen name="Detail" component={Detail} />
        </Stack.Navigator>
      </NavigationContainer>
    </QueryClientProvider>
  );
}
