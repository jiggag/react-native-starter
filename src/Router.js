import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import { useSelector } from 'react-redux';
import Root from './pages/Root';
import SignUp from './pages/SignUp';

const Stack = createStackNavigator();

const Router = () => {
  const { token } = useSelector(({ auth }) => auth);

  return (
    <NavigationContainer>
      <Stack.Navigator headerMode="none">
        {token ? (
          <Stack.Screen name="Root" component={Root} />
        ) : (
          <Stack.Screen name="SignUp" component={SignUp} />
        )}
      </Stack.Navigator>
    </NavigationContainer>
  );
};

export default Router;
