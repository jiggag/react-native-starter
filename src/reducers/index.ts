import { enableES5 } from 'immer';
import { combineReducers } from 'redux';
import auth, { AuthReducer } from './auth';
import root, { RootReducer } from './root';

interface ReducerProps {
  root: RootReducer;
  auth: AuthReducer;
}

enableES5();

export default combineReducers<ReducerProps>({
  root,
  auth,
});
