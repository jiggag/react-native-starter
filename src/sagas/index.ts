import { all, fork } from 'redux-saga/effects';
import _map from 'lodash/map';
import auth from './auth';
import root from './root';

const sagas = [...auth, ...root];

export default function* rootSagas() {
  try {
    yield all(_map(sagas, (saga) => fork(saga)));
  } catch (e) {
    console.error(e);
  }
}
