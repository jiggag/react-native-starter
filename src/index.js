import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import rootReducer from './reducers';
import Router from './Router';

const store = createStore(rootReducer);

const App = () => {
  return (
    <Provider store={store}>
      <Router />
    </Provider>
  );
};

export default App;
