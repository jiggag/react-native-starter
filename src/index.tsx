import React from 'react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga';
import { PersistGate } from 'redux-persist/integration/react';
import { persistStore, persistReducer } from 'redux-persist';
import AsyncStorage from '@react-native-community/async-storage';
import rootReducer from './reducers';
import saga from './sagas';
import Router from './Router';

const sagaMiddleware = createSagaMiddleware();
const middlewares = [sagaMiddleware];

if (__DEV__) {
  const createDebugger = require('redux-flipper').default;
  middlewares.push(createDebugger());
}

const persistConfig = {
  key: 'root',
  storage: AsyncStorage,
  whitelist: [],
};
const persistedReducer = persistReducer(persistConfig, rootReducer);

const store = createStore(persistedReducer, applyMiddleware(...middlewares));
const persistor = persistStore(store);
sagaMiddleware.run(saga);

const App = () => {
  return (
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <Router />
      </PersistGate>
    </Provider>
  );
};

export default App;
