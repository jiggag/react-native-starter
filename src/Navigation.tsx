import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import { SafeAreaProvider } from 'react-native-safe-area-context';
import { useSelector } from 'react-redux';
import { ReducerProps } from 'reducers';
import Root from './pages/Root';
import SignUp from './pages/SignUp';
import Splash from './pages/Splash';

const Stack = createStackNavigator();

export const Navigation = () => {
  const {
    auth: { token },
    root: { isAppSplash },
  } = useSelector((state: ReducerProps) => state);

  return (
    <SafeAreaProvider>
      {isAppSplash ? (
        <Splash />
      ) : (
        <NavigationContainer>
          <Stack.Navigator headerMode="none">
            {token ? <Stack.Screen name="Root" component={Root} /> : <Stack.Screen name="SignUp" component={SignUp} />}
          </Stack.Navigator>
        </NavigationContainer>
      )}
    </SafeAreaProvider>
  );
};
