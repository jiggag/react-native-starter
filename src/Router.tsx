import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import { useSelector } from 'react-redux';
import { ReducerProps } from 'reducers';
import Root from './pages/Root';
import SignUp from './pages/SignUp';
import Splash from './pages/Splash';

const Stack = createStackNavigator();

const Router = () => {
  const {
    auth: { token },
    root: { isAppSplash },
  } = useSelector((state: ReducerProps) => state);

  return isAppSplash ? (
    <Splash />
  ) : (
    <NavigationContainer>
      <Stack.Navigator headerMode="none">
        {token ? <Stack.Screen name="Root" component={Root} /> : <Stack.Screen name="SignUp" component={SignUp} />}
      </Stack.Navigator>
    </NavigationContainer>
  );
};

export default Router;
