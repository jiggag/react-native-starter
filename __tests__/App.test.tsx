import 'react-native';
import React from 'react';
import { render } from '@testing-library/react-native';
import { App } from '../src/App';

jest.mock('@notifee/react-native', () => ({
  requestPermission: jest.fn(),
  onForegroundEvent: jest.fn(),
}));

it('renders correctly', () => {
  const { toJSON } = render(<App />);
  expect(toJSON()).toMatchSnapshot();
});
