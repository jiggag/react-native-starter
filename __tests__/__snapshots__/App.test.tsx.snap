// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      nativeID="animatedComponent"
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Main"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            undefined,
          ]
        }
      >
        <RCTSafeAreaView
          emulateUnlessSupported={true}
        >
          <RCTScrollView
            contentInsetAdjustmentBehavior="automatic"
          >
            <View>
              <Text
                style={
                  Object {
                    "fontSize": 24,
                    "fontWeight": "600",
                  }
                }
              >
                Starter (Main)
              </Text>
              <View
                accessibilityRole="button"
                accessible={true}
                collapsable={false}
                focusable={true}
                nativeID="animatedComponent"
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {},
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#007AFF",
                          "fontSize": 18,
                          "margin": 8,
                          "textAlign": "center",
                        },
                      ]
                    }
                  >
                    Go Detail
                  </Text>
                </View>
              </View>
            </View>
          </RCTScrollView>
        </RCTSafeAreaView>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
